pkg/controllers/monitoring-stack/controller.go

```diff
@@ -137,15 +143,76 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		// retrying after some time.
 		if errors.IsAlreadyExists(err) || errors.IsConflict(err) {
 			logger.V(3).Info("skipping reconcile error", "err", err)
+			failedCondition := newFailedCondition(available, err, ms.ObjectMeta.Generation)
+			err := r.updateStatusCondition(ctx, ms, failedCondition)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
 			return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
 		}
 		if err != nil {
+			failedCondition := newFailedCondition(available, err, ms.ObjectMeta.Generation)
+			err := r.updateStatusCondition(ctx, ms, failedCondition)
```

Comment:
I think we can simplify this to `return r.updateStatusCondition(ctx, ms, failedCondition)` by making `r.updateStatusCondition` return `ctrl.Result, error`

Reply:
outdated